### PR TITLE
[ExportVerilog] Improve XMR emission when used as bound inputs

### PR DESCRIPTION
**Pull request by uenoku**
For https://github.com/llvm/circt/issues/3853, this commit improves emission of XMR to inline XMR references into bound instance ports. 

For the example in #3853, the difference of w/ and w/o the PR:
```verilog
<   wire struct packed {logic x; logic y; } _TopXMRAsserts_inst0_a;
<   wire                                    _TopXMRAsserts_inst0_b;
38,39d35
<   assign _TopXMRAsserts_inst0_a = middle.bottom.O;
<   assign _TopXMRAsserts_inst0_b = middle.bottom.I.x;
44,45c40,41
<       .a (_TopXMRAsserts_inst0_a),
<       .b (_TopXMRAsserts_inst0_b)
---
>       .a (middle.bottom.O),
>       .b (middle.bottom.I.x)
57,58c53,54
<   .a (_TopXMRAsserts_inst0_a),
<   .b (_TopXMRAsserts_inst0_b)
---
>   .a (middle.bottom.O),
>   .b (middle.bottom.I.x)
```

CC: @rsetaluri

### CODE REVIEW

File: lib/Conversion/ExportVerilog/PrepareForEmission.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ bool ExportVerilog::isSimpleReadOrPort(Value v) {
   auto readSrc = read.getInput().getDefiningOp();
   if (!readSrc)
     return false;
-  return isa<WireOp, RegOp>(readSrc);
+  return isa<WireOp, RegOp, LogicOp, XMROp>(readSrc);
```
Comment (seldridge) [inline lib/Conversion/ExportVerilog/PrepareForEmission.cpp:52]: The addition of `LogicOp` seems unrelated to this PR?

**Review by seldridge — APPROVED** [review]
This looks great!

File: lib/Conversion/ExportVerilog/PrepareForEmission.cpp (hunk 1 above)
Comment (uenoku) [reply lib/Conversion/ExportVerilog/PrepareForEmission.cpp:52]: Right, unrelated (I found I was missing LogicOp either :). I'll land separately.
